pkg/operator/operator.go

```diff
@@ -90,32 +98,52 @@ type Operator struct {
 }
 
 func NewOperator(ctx context.Context, operator *operator.Operator) (context.Context, *Operator) {
-	config := &aws.Config{
-		STSRegionalEndpoint: endpoints.RegionalSTSEndpoint,
+	cfg, err := config.LoadDefaultConfig(ctx,
+		config.WithSTSRegionalEndpoint(endpoints.RegionalSTSEndpoint),
+		config.WithRetryer(func() aws.Retryer {
+			return retry.NewStandard(func(o *retry.StandardOptions) {
+				o.MaxAttempts = 5
+			}),
+		}),
+		config.WithAPIOptions(
+			middleware.AddUserAgentKey("CustomUserAgent"),
+			prometheusv1.WithPrometheusMetrics(crmetrics.Registry),
+		),
+	)
+	if err != nil {
+		log.Fatalf("unable to load SDK config, %v", err)
 	}
 
 	// prometheusv1.WithPrometheusMetrics is used until the upstream aws-sdk-go or aws-sdk-go-v2 supports
 	// Prometheus metrics for client-side metrics out-of-the-box
 	// See: https://github.com/aws/aws-sdk-go-v2/issues/1744
-	sess := prometheusv1.WithPrometheusMetrics(WithUserAgent(session.Must(session.NewSession(
-		request.WithRetryer(
-			config,
-			awsclient.DefaultRetryer{NumMaxRetries: awsclient.DefaultRetryerMaxNumRetries},
-		),
-	))), crmetrics.Registry)
-
-	if *sess.Config.Region == "" {
+	if cfg.Region == "" {
 		log.FromContext(ctx).V(1).Info("retrieving region from IMDS")
-		region, err := ec2metadata.New(sess).Region()
-		*sess.Config.Region = lo.Must(region, err, "failed to get region from metadata server")
+		metadataClient := ec2metadata.NewEC2Metadata(cfg)
+		region, err := metadataClient.Region(ctx)
+		if err != nil {
+			log.FromContext(ctx).Error(err, "failed to get region from metadata server")
+			os.Exit(1)
+		}
+		cfg.Region = region
 	}
-	ec2api := ec2.New(sess)
+
+	ec2api := ec2.NewFromConfig(cfg)
+	eksapi := eks.NewFromConfig(cfg)
+	iamapi := iam.NewFromConfig(cfg)
+	pricingapi := pricing.NewFromConfig(cfg)
+	ssmapi := ssm.NewFromConfig(cfg)
+
+	// Check EC2 connectivity
+	
 	if err := CheckEC2Connectivity(ctx, ec2api); err != nil {
 		log.FromContext(ctx).Error(err, "ec2 api connectivity check failed")
 		os.Exit(1)
 	}
-	log.FromContext(ctx).WithValues("region", *sess.Config.Region).V(1).Info("discovered region")
-	clusterEndpoint, err := ResolveClusterEndpoint(ctx, eks.New(sess))
+	log.Printf("discovered region: %s\n", cfg.Region)
```

Comment:
We should keep the old form of our logger here -- I think this may have been an unintentional change